test/e2e/specs/editor/blocks/navigation.spec.js

```diff
@@ -802,6 +802,500 @@ test.describe( 'Navigation block', () => {
+		test( 'overlay menu opens on click on open menu button', async ( {
+			page,
+		} ) => {
+			const overlayMenu = page.locator(
+				'nav.wp-block-navigation div.has-modal-open'
+			);
+			const openMenuButton = page.locator( '[aria-label="Open menu"]' );
+			await expect( overlayMenu ).toBeHidden();
+			await openMenuButton.click();
+			await expect( overlayMenu ).toBeVisible();
+		} );
+
+		test( 'overlay menu closes on click on close menu button', async ( {
+			page,
+		} ) => {
+			const overlayMenu = page.locator(
+				'nav.wp-block-navigation div.has-modal-open'
+			);
+			const openMenuButton = page.locator( '[aria-label="Open menu"]' );
+			const closeMenuButton = page.locator( '[aria-label="Close menu"]' );
+			await expect( overlayMenu ).toBeHidden();
+			await openMenuButton.click();
+			await expect( overlayMenu ).toBeVisible();
+			await closeMenuButton.click();
+			await expect( overlayMenu ).toBeHidden();
+		} );
+
+		test( 'overlay menu closes on ESC key', async ( { page } ) => {
+			const overlayMenu = page.locator(
+				'nav.wp-block-navigation div.has-modal-open'
+			);
+			const openMenuButton = page.locator( '[aria-label="Open menu"]' );
+			await expect( overlayMenu ).toBeHidden();
+			await openMenuButton.click();
+			await expect( overlayMenu ).toBeVisible();
+			await page.keyboard.press( 'Escape' );
+			await expect( overlayMenu ).toBeHidden();
+			await expect( openMenuButton ).toBeFocused();
+		} );
+
+		test( 'overlay menu focuses on first element after opening', async ( {
+			page,
+		} ) => {
+			const overlayMenu = page.locator(
+				'nav.wp-block-navigation div.has-modal-open'
+			);
+			const openMenuButton = page.locator( '[aria-label="Open menu"]' );
+			const firstItem = page.locator(
+				'ul li.wp-block-navigation-item:first-child > *:first-child'
+			);
+			await expect( overlayMenu ).toBeHidden();
+			await openMenuButton.click();
+			await expect( overlayMenu ).toBeVisible();
+			await expect( firstItem ).toBeFocused();
+		} );
+
+		test( 'overlay menu traps focus', async ( { page } ) => {
+			const overlayMenu = page.locator(
+				'nav.wp-block-navigation div.has-modal-open'
+			);
+			const openMenuButton = page.locator( '[aria-label="Open menu"]' );
+			const firstItem = page.locator(
+				'ul li.wp-block-navigation-item:first-child > *:first-child'
+			);
+			const closeMenuButton = page.locator( '[aria-label="Close menu"]' );
+			await expect( overlayMenu ).toBeHidden();
+			await openMenuButton.click();
+			await expect( overlayMenu ).toBeVisible();
+			await expect( firstItem ).toBeFocused();
+			await page.keyboard.press( 'Tab' );
+			await page.keyboard.press( 'Tab' );
+			await expect( closeMenuButton ).toBeFocused();
+			await page.keyboard.press( 'Shift+Tab' );
+			await page.keyboard.press( 'Shift+Tab' );
+			await expect( firstItem ).toBeFocused();
+		} );
+	} );
+
+	test.describe( 'Submenus (Open on click)', () => {
+		test.beforeEach( async ( { admin, editor, page, requestUtils } ) => {
```

Comment:
Same remarks as in the previous describe subsection:
- beforeAll is better
- and navigation to the page `/` could be moved to each test